Session08/binarytree/solutions/BinarySearchTree.java

```diff
@@ -0,0 +1,123 @@
+// This class fails to compile, can you figure out why, and fix it?
+// This is beacuse BinaryTree has a constructor which is not default. Hence, when we extend this class and try to call a no-argument constructor, it'll give a compile-time error. so, we call it by passing the parameters (to instantiate the root node) in super() called inside the constructor of BinarySearchTree
+
+class BinarySearchTree extends BinaryTree {
+
+    private int flag = 0;
+    
+    public BinarySearchTree()
+    {
+        super(0);
+    }
+    
+    // insert node
+	@Override
+	public boolean insert(int value) {
+		super.root = insert(super.root, value);
```

Comment:
What if the node already exists? This function always returns true regardless of whether the insert "fails" or succeeds.